### ISSUE TITLE
EBS volume_type ignored

### ISSUE THREAD

**Issue by jfindley** [ansible/ansible-modules-core#2040]
## Issue type

Bug report
## Summary

When creating an ec2 volume, volume_type is ignored.

The config:

``` yaml
volumes:
  device_name: /dev/sdb
        volume_size: 100
        volume_type: gp2
```

creates a volume using standard disks, not gp2.

The config:

``` yaml
volumes:
  device_name: /dev/sdb
        volume_size: 100
        volume_type: io1
        iops: 150
```

produces an error message: "Instance creation failed => InvalidParameterCombination: The parameter iops is not supported for standard volumes."
## Ansible version:

```
ansible --version
ansible 2.0.0 (detached HEAD 43cc27084d) last updated 2015/08/13 15:18:18 (GMT +100)
  lib/ansible/modules/core: (devel 1c7e8ef1a0) last updated 2015/09/09 17:21:32 (GMT +100)
  lib/ansible/modules/extras: (detached HEAD ff2386faf4) last updated 2015/07/21 18:18:28 (GMT +100)
  v1/ansible/modules/core: (detached HEAD f8d8af17cd) last updated 2015/07/21 18:18:36 (GMT +100)
  v1/ansible/modules/extras: (detached HEAD 495ad450e5) last updated 2015/07/21 18:18:40 (GMT +100)
  config file = /etc/ansible/ansible.cfg
  configured module search path = None
```
## Environment/config:

OSX source install.  Default config.

**Comment by evanccnyc** [comment]
This appears to be because you are using volume_type instead of device_type under the volumes settings. Why volume_type is allowed there I dont know.

**Comment by evanccnyc** [comment]
If anyone wanted to we could add the following lines around 729:

```
    if 'volume_type' in volume:
        volume_type = volume.get('volume_type')
    elif 'device_type' in volume:
        volume_type = volume.get('device_type')
    return BlockDeviceType(snapshot_id=volume.get('snapshot'),
                           ephemeral_name=volume.get('ephemeral'),
                           size=volume.get('volume_size'),
                           volume_type=volume_type,
                           delete_on_termination=volume.get('delete_on_termination', False),
                           iops=volume.get('iops'),
                           encrypted=volume.get('encrypted', None))
```

Not sure its worth it though

**Comment by evanccnyc** [comment]
The bigger problem is that ec2_vol uses 'volume_type' where as ec2 and ec2_lc uses 'device_type'

**Comment by willthames** [comment]
To be fair, ec2 was first to incorporate device_type as a setting. But no idea why I chose to add it as device_type when BlockDeviceType takes volume_type.

I have no objection to the above 'fix' (and perhaps marking device_type as deprecated in 2.0) 

could just do 
`volume_type=volume.get('volume_type', volume.get('device_type'))` 
which is a much smaller fix. And update the docs to use volume_type.

**Comment by evanccnyc** [comment]
We should probably cc @bcoca and others who might want to weigh in on the lack of consistency in options.

**Comment by bcoca** [comment]
This was missed (I really hate these inconsistencies), but at this point I'm thinking we should just alias it in both places so they are equivalent.

**Comment by willthames** [comment]
It's not really an option though @bcoca, so you can't really alias it.

It's a parameter to a dictionary option (i.e. `volumes` is the option, `device_type` (or `volume_type`) is a dictionary key that belongs to a dictionary key that belongs to `volumes`.

There is no aliasing of those, at least not at present, that is allowed by Ansible - you either need to make that a thing, or just handle it in a manner similar to what @evanccnyc suggests

**Comment by bcoca** [comment]
I'm aware we cannot use the spec alias, a `.get` or `if 'device_type' in` was what i was thinking of.

**Comment by evanccnyc** [comment]
This is fixed.

**Comment by bcoca** [comment]
closing via merged #2124
